Library/Formula/hbase.rb

```diff
@@ -4,32 +4,151 @@ class Hbase < Formula
   url "https://www.apache.org/dyn/closer.cgi?path=hbase/1.1.2/hbase-1.1.2-bin.tar.gz"
   sha256 "8ca5bf0203cef86b4a0acbba89afcd5977488ebc73eec097e93c592b16f8bede"
 
-  bottle :unneeded
-
   depends_on :java => "1.6+"
-  depends_on "hadoop"
+  depends_on "hadoop" => :optional
+  depends_on "lzo" => :recommended
+  depends_on "ant" => :build if build.with? "lzo"
+
+  resource "hadoop-lzo" do
+    url "https://github.com/cloudera/hadoop-lzo/archive/0.4.14.tar.gz"
+    sha256 "aa8ddbb8b3f9e1c4b8cc3523486acdb7841cd97c002a9f2959c5b320c7bb0e6c"
+  end
 
   def install
+    ENV.java_cache if build.with? "lzo"
     rm_f Dir["bin/*.cmd", "conf/*.cmd"]
     libexec.install %w[bin conf docs lib hbase-webapps]
     bin.write_exec_script Dir["#{libexec}/bin/*"]
 
-    inreplace "#{libexec}/conf/hbase-env.sh",
-      "# export JAVA_HOME=/usr/java/jdk1.6.0/",
-      "export JAVA_HOME=\"$(/usr/libexec/java_home)\""
+    if build.with? "lzo"
+      resource("hadoop-lzo").stage do
+        # Fixed upstream: https://github.com/cloudera/hadoop-lzo/blob/master/build.xml#L235
+        inreplace "build.xml",
+                  %r{(<class name="com.hadoop.compression.lzo.LzoDecompressor" />)},
+                  "\\1\n<classpath refid=\"classpath\"/>"
+        ENV["CLASSPATH"] = Dir["#{libexec}/lib/hadoop-common-*.jar"].first
+        ENV["CFLAGS"] = "-m64"
```

Comment:
What happens if the `-m64` flag is just omitted? This sounds like it's maybe working around a shortcoming in `hadoop-lzo`'s `configure`, and not an actual platform requirement. The [hadoop-lzo README building & configuring section](https://github.com/twitter/hadoop-lzo/tree/f761f10a818fa286fbad0466c3279da6739a01ff#building-and-configuring) says it is "built exactly the same way" as hadoop-gpl, whose [FAQ](https://code.google.com/archive/a/apache-extras.org/p/hadoop-gpl-compression/wikis/FAQ.wiki) says they also support 32-bit builds using `-m32`. Let's see if we can tweak the hadoop-lzo build process to autodetect the architecture. Or if the user-provided flag is really necessary, we could throw in `-m32` or `-m64` conditioned on `OS::Mac::preferred_arch` or similar.